cirq/ops/linear_combinations.py

```diff
@@ -315,6 +316,90 @@ def copy(self) -> 'PauliSum':
         factory = type(self)
         return factory(self._linear_dict.copy())
 
+    def expectation_from_wavefunction(
+            self,
+            state: np.ndarray,
+            qubit_map: Dict[raw_types.Qid, int]) -> float:
+        """Evaluate the expectation of this PauliSum given a wavefunction.
+
+        See `PauliString.expectation_from_wavefunction`.
+        """
+        if any([abs(p.coefficient.imag) > 0.0001 for p in self]):
+            raise NotImplementedError(
+                "Cannot compute expectation value of a non-Hermitian "
+                "PauliString <{}>. Coefficient must be real.".format(self))
+
+        if not isinstance(qubit_map, dict) or not all(
+            isinstance(k, raw_types.Qid) and isinstance(v, int) for k, v in qubit_map.items()):
+            raise TypeError("Input qubit map must be a valid mapping from "
+                            "Qubit ID's to integer indices.")
+
+        if set(qubit_map.keys()) < set(self.qubits):
```

Comment:
You want `not set(self.qubits) >= qubit_map.keys()`. For example, suppose the sets are disjoint.

Add tests exercising this case for A>B, A<B, and mutual-subset-but-different-extra-items.

Reply:
done.